src/include/distributed/resource_lock.h

```diff
@@ -125,6 +128,33 @@ typedef enum CitusOperations
 						 (uint32) 0, \
 						 ADV_LOCKTAG_CLASS_CITUS_LOGICAL_REPLICATION)
 
+/* advisory lock for nonblocking shard splitting, also it has the database hardcoded to MyDatabaseId,
+ * to ensure the locks are local to each database */
+#define SET_LOCKTAG_NONBLOCKING_SPLIT(tag) \
+	SET_LOCKTAG_ADVISORY(tag, \
+						 MyDatabaseId, \
+						 (uint32) 0, \
+						 (uint32) 0, \
+						 ADV_LOCKTAG_CLASS_CITUS_NONBLOCKING_SPLIT)
+
+/* advisory lock for creating distributed table concurrently, also it has the database hardcoded to MyDatabaseId,
+ * to ensure the locks are local to each database */
+#define SET_LOCKTAG_CREATE_DISTRIBUTED_TABLE_CONCURRENTLY(tag) \
```

Comment:
minor: we have this CitusOperations enum and locktag, which seems designed for these types of scenarios